File: android-native/app/src/main/java/com/example/kpkn/screens/workout/WorkoutLoadSuggestionRules.kt
package com.example.kpkn.screens.workout

@Deprecated(
    message = "Use LoadSuggestionEngine.fatigueFactorForPriorCompletedSets",
    replaceWith = ReplaceWith(
        "LoadSuggestionEngine.fatigueFactorForPriorCompletedSets(priorCompletedCount)",
        "com.example.kpkn.domain.workout.LoadSuggestionEngine",
    ),
)
object WorkoutLoadSuggestionRules {
    fun fatigueFactorForPriorCompletedSets(priorCompletedCount: Int): Double =
        com.example.kpkn.domain.workout.LoadSuggestionEngine.fatigueFactorForPriorCompletedSets(priorCompletedCount)
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Screens/Workout/WorkoutLoadSuggestionRules.swift
import Foundation

struct WorkoutLoadSuggestionRules {
    static func fatigueFactorForPriorCompletedSets(priorCompletedCount: Int) -> Double {
        switch max(0, priorCompletedCount) {
        case 0: return 1.0
        case 1: return 0.8
        case 2: return 0.6
        default: return 0.5
        }
    }
}
